### PR TITLE
Add section to emoji keyboard, move emoji in plist

### PR DESCRIPTION
**Pull request by daehn**
What's in this PR?

* The emoji are now separated in different sections
* Each section is stored in a plist file for easy editing by design etc.
* The `EmojiCollectionView` is sectioned as well and a new class `EmojiSectionViewController` is used to display the icons at the bottom for quick selection and panning between sections.
* A backspace button has been added to the emoji keyboard.

#### What's not in this PR?

* The "most used" section is not yet implemented and will come in a separate PR 

bildschirmfoto 2016-09-29 um 16 07 50

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/EmojiSectionViewController.swift [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+protocol EmojiSectionViewControllerDelefate: class {
+    func sectionViewControler(_ viewController: EmojiSectionViewController, performAction action: EmojiSectionViewController.Action)
+}
+
+
+class EmojiSectionViewController: UIViewController {
+    
+    enum Action {
+        case Select(EmojiSectionType)
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/EmojiSectionViewController.swift]: I think in Swift 3 it is preferred to have a camel-case enum elements name.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emoji/EmojiKeyboardViewController.swift [hunk 2]
```diff
@@ -23,66 +23,93 @@ import Cartography
...
     
-    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: EmojiCollectionViewCell.zm_reuseIdentifier, for: indexPath) as! EmojiCollectionViewCell
-        cell.titleLabel.text = emojis[indexPath.item]
-        return cell
+}
+
+extension EmojiKeyboardViewController: EmojiSectionViewControllerDelefate {
+
+    func sectionViewControler(_ viewController: EmojiSectionViewController, performAction action: EmojiSectionViewController.Action) {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emoji/EmojiKeyboardViewController.swift]: typo: `sectionViewControler` > `sectionViewController`

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emoji/EmojiKeyboardViewController.swift [hunk 3]
```diff
@@ -23,66 +23,93 @@ import Cartography
...
+        return cell
     }
     
-    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: EmojiCollectionViewCell.zm_reuseIdentifier, for: indexPath) as! EmojiCollectionViewCell
-        cell.titleLabel.text = emojis[indexPath.item]
-        return cell
+}
+
+extension EmojiKeyboardViewController: EmojiSectionViewControllerDelefate {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emoji/EmojiKeyboardViewController.swift]: typo: `EmojiSectionViewControllerDelefate` > `EmojiSectionViewControllerDelegate `

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/EmojiSectionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+import Foundation
+import Cartography
+
+
+protocol EmojiSectionViewControllerDelefate: class {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/EmojiSectionViewController.swift]: typo: `EmojiSectionViewControllerDelefate` > `EmojiSectionViewControllerDelegate `

File: WireExtensionComponents/Views/Buttons/IconButton.h [hunk 5]
```diff
@@ -43,3 +46,5 @@
 - (void)setTitleImageSpacing:(CGFloat)titleImageSpacing horizontalMargin:(CGFloat)horizontalMargin;
 
 @end
+
+NS_ASSUME_NONNULL_END
```
Comment (mikeger) [inline WireExtensionComponents/Views/Buttons/IconButton.h]: I think `-iconColorForState:` can return nil

**Pull request merged**